api/v1alpha1/external_secrets_types.go

```diff
@@ -0,0 +1,195 @@
+	// +kubebuilder:validation:Optional
+	// +mapType=atomic
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+}
+
+// ControllerConfig is for configuring the operator for setting up
+// defaults to install external-secrets.
+type ControllerConfig struct {
+	// namespace is for configuring the namespace to install the external-secret operand.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:="external-secrets"
+	Namespace string `json:"namespace,omitempty"`
+
+	// labels to apply to all resources created for external-secrets deployment.
+	// +mapType=granular
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// BitwardenSecretManagerProvider is for enabling the bitwarden secrets manager provider and
+// for setting up the additional service required for connecting with the bitwarden server.
+type BitwardenSecretManagerProvider struct {
+	// enabled is for enabling the bitwarden secrets manager provider, which can be indicated
+	// by setting `true` or `false`.
+	// +kubebuilder:default:="false"
+	// +kubebuilder:validation:Enum:="true";"false"
+	// +kubebuilder:validation:Optional
+	Enabled string `json:"enabled,omitempty"`
+}
+
+// WebhookConfig is for configuring external-secrets webhook specifics.
+type WebhookConfig struct {
+	// CertificateCheckInterval is for configuring the polling interval to check the certificate
+	// validity.
+	// +kubebuilder:default:="5m"
+	// +kubebuilder:validation:Optional
+	CertificateCheckInterval metav1.Duration `json:"certificateCheckInterval,omitempty"`
+
+	// CertManagerConfig is for configuring cert-manager specifics.
+	// +kubebuilder:validation:Optional
+	CertManagerConfig *CertManagerConfig `json:"certManagerConfig,omitempty"`
+}
+
+// CertManagerConfig is for configuring cert-manager specifics.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.issuerRef) && !has(self.issuerRef) || has(oldSelf.issuerRef) && has(self.issuerRef)",message="issuerRef may only be configured during creation"
+type CertManagerConfig struct {
+	// enabled is for enabling the use of cert-manager for obtaining and renewing the
+	// certificates used for webhook server, instead of built-in certificates.
+	// Use `true` or `false` to indicate the preference.
+	// +kubebuilder:default:="false"
+	// +kubebuilder:validation:Enum:="true";"false"
+	// +kubebuilder:validation:Optional
+	Enabled string `json:"enabled,omitempty"`
+
+	// addInjectorAnnotations is for adding the `cert-manager.io/inject-ca-from` annotation to the
+	// webhooks and CRDs to automatically setup webhook to the cert-manager CA. This requires
+	// CA Injector to be enabled in cert-manager. Use `true` or `false` to indicate the preference.
+	// +kubebuilder:default:="false"
+	// +kubebuilder:validation:Enum:="true";"false"
+	// +kubebuilder:validation:Optional
+	AddInjectorAnnotations string `json:"addInjectorAnnotations,omitempty"`
```

Comment:
How about making this a `bool` instead?

Reply:
My understanding is that, we should avoid using bool types in APIs. I am unable to recall the reason, this came up during previous API reviews, guess it as something to do with kubebuilder validations.